kitsune/sumo/middleware.py

```diff
@@ -78,6 +80,13 @@ class LocaleURLMiddleware(object):
     """
 
     def process_request(self, request):
+        try:
```

Comment:
Can we remove the try/except from here? If ti's not possible let's rewrite this to remove some logic out of try clause.  How about using request.path?